usr/bin/regolith-session-init

```diff
@@ -5,28 +5,47 @@ set -Eeu -o pipefail
 
 source /usr/lib/regolith/regolith-session-common.sh
 
-# --------- Script Main -----------
-
-load_standard_xres  # ~/.Xresources
-load_regolith_xres  # ~/.config/regolith2/Xresources
-
-# Register with gnome-session so that it does not kill the whole session thinking it is dead.
-# See https://zork.net/~st/jottings/gnome-i3.html for details
-if [ -n "$DESKTOP_AUTOSTART_ID" ]; then
-    dbus-send --print-reply --session --dest=org.gnome.SessionManager "/org/gnome/SessionManager" org.gnome.SessionManager.RegisterClient "string:Regolith" "string:$DESKTOP_AUTOSTART_ID"
-fi
+start_i3 () {
+    load_standard_xres  # ~/.Xresources
+    load_regolith_xres  # ~/.config/regolith2/Xresources
+
+    # Register with gnome-session so that it does not kill the whole session thinking it is dead.
+    # See https://zork.net/~st/jottings/gnome-i3.html for details
+    if [ -n "$DESKTOP_AUTOSTART_ID" ]; then
+        dbus-send --print-reply --session --dest=org.gnome.SessionManager "/org/gnome/SessionManager" org.gnome.SessionManager.RegisterClient "string:Regolith" "string:$DESKTOP_AUTOSTART_ID"
+    fi
+
+    echo "Regolith is launching i3 with $I3_CONFIG_FILE"
+    i3 -c "$I3_CONFIG_FILE"
+
+    # Run user's post logout script if script. Post logout means that the X session is closed at this point.
+    if [ -f "$USER_POST_LOGOUT_SCRIPT_FILE" ] ; then
+        echo "Regolith is launching user post-logout file $USER_POST_LOGOUT_SCRIPT_FILE"
+        timeout --verbose 5 bash "$USER_POST_LOGOUT_SCRIPT_FILE"
+    fi
+
+    # Close session when i3 exits.
+    if [ -n "$DESKTOP_AUTOSTART_ID" ]; then
+        dbus-send --print-reply --session --dest=org.gnome.SessionManager "/org/gnome/SessionManager" org.gnome.SessionManager.Logout "uint32:1"
+    fi
+}
+
+start_sway () {
+    load_standard_trawlres
+    load_regolith_trawlres
+    
+    echo "Regolith is launching sway with $SWAY_CONFIG_FILE"
+    sway -c "$SWAY_CONFIG_FILE"
+}
 
+# --------- Script Main -----------
 resolve_default_config_file
-echo "Regolith is launching i3 with $I3_CONFIG_FILE"
-i3 -c "$I3_CONFIG_FILE"
-
-# Run user's post logout script if script. Post logout means that the X session is closed at this point.
-if [ -f "$USER_POST_LOGOUT_SCRIPT_FILE" ] ; then
-    echo "Regolith is launching user post-logout file $USER_POST_LOGOUT_SCRIPT_FILE"
-	timeout --verbose 5 bash "$USER_POST_LOGOUT_SCRIPT_FILE"
-fi
-
-# Close session when i3 exits.
-if [ -n "$DESKTOP_AUTOSTART_ID" ]; then
-    dbus-send --print-reply --session --dest=org.gnome.SessionManager "/org/gnome/SessionManager" org.gnome.SessionManager.Logout "uint32:1"
+export MOZ_ENABLE_WAYLAND=1
```

Comment:
Why is this needed?

Reply:
In my experience, without this Firefox will use legacy playback methods (as in hardware acceleration won't work). This leads to unacceptable CPU usage and occasional jittering when playing in 4k.

Reply:
There's also the added benefit of support for touchpad gestures (like pinch to zoom) and maybe a few other features that can only work in Wayland.

Reply:
As mentioned, I think this belongs elsewhere or possibly just in documentation.